app/src/main/java/com/honeybeard/androidfundamentals/activities/movie_list/MoviesListFragment.kt

```diff
@@ -0,0 +1,69 @@
+package com.honeybeard.androidfundamentals.activities.movie_list
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.honeybeard.androidfundamentals.R
+import com.honeybeard.androidfundamentals.activities.main.MainActivity
+
+// TODO: Rename parameter arguments, choose names that match
+// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+private const val ARG_PARAM1 = "param1"
+private const val ARG_PARAM2 = "param2"
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [MoviewListFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+class MoviewListFragment : Fragment() {
+    // TODO: Rename and change types of parameters
+    private var param1: String? = null
+    private var param2: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            param1 = it.getString(ARG_PARAM1)
+            param2 = it.getString(ARG_PARAM2)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_movies_list, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        view.findViewById<View>(R.id.movie_preview).setOnClickListener{
+            (activity as MainActivity).selectMovie()
```

Comment:
here your fragment directly depend to mainActivity 
**OPTIONAL** hint that will be good if you do it via interface
here don't need direct call 
1)need to create Interface with abstract `selectMovie` method
2)implement it to mainActivity
3)inside fragment `onAttach` get reference to this fragment
4)and `onDetach` clear reference